Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -0,0 +1,272 @@
+                UnregisterPowerShellStreams(streamHandler);
+                _runspaceSemaphore.Release();
+            }
+        }
+
+        private async Task HandleRequestCompletion(
+            RequestCompletion requestCompletion,
+            KernelInvocationContext context)
+        {
+            var completionRequestReceived = new CompletionRequestReceived(requestCompletion);
+
+            context.Publish(completionRequestReceived);
+
+            var completionList =
+                await GetCompletionList(
+                    requestCompletion.Code, 
+                    requestCompletion.CursorPosition);
+
+            context.Publish(new CompletionRequestCompleted(completionList, requestCompletion));
+        }
+
+        private Task HandleCancelCurrentCommand(
+            CancelCurrentCommand cancelCurrentCommand,
+            KernelInvocationContext context)
+        {
+            lock (_cancellationSourceLock)
+            {
+                _cancellationSource.Cancel();
+                _cancellationSource = new CancellationTokenSource();
+                if (_pwsh.Runspace.RunspaceAvailability != RunspaceAvailability.Available)
+                {
+                    _pwsh.Stop();
+                }
+            }
+
+            var reply = new CurrentCommandCancelled(cancelCurrentCommand);
+            context.Publish(reply);
+
+            return Task.CompletedTask;
+        }
+
+        #endregion
+
+        public bool IsCompleteSubmission(string code)
+        {
+            // Parse the PowerShell script. If there are any parse errors, then we return false.
+            Parser.ParseInput(code, out Token[] tokens, out ParseError[] errors);
+            return errors.Length > 0;
+        }
+
+        private async Task<IEnumerable<CompletionItem>> GetCompletionList(
+            string code,
+            int cursorPosition)
+        {
+            await _runspaceSemaphore.WaitAsync();
+
+            try
+            {
+                CommandCompletion completion = CommandCompletion.CompleteInput(code, cursorPosition, null, _pwsh);
+
+                return completion.CompletionMatches.Select(c => new CompletionItem(
+                    displayText: c.CompletionText,
+                    kind: c.ResultType.ToString(),
+                    documentation: c.ToolTip
+                ));
+            }
+            finally
+            {
+                _runspaceSemaphore.Release();
+            }
+        }
+
+        public async Task LoadExtensionsFromDirectory(
+            DirectoryInfo directory,
+            KernelInvocationContext context)
+        {
+            var extensionsDirectory =
+                new DirectoryInfo(
+                    Path.Combine(
+                        directory.FullName,
+                        "interactive-extensions",
+                        "dotnet",
+                        "cs"));
+
+            await new KernelExtensionAssemblyLoader().LoadFromAssembliesInDirectory(
```

Comment:
Why not have a `_extensionLoader` and initialize it in the constructor like the CSharp kernel?